### ISSUE TITLE
Nightly tests of MappingApplication fail

### ISSUE THREAD

**Issue by philbucher** [KratosMultiphysics/Kratos#1037]
The nightly tests of the MappingApplication fail with the following error:
`Running tests for MappingApplication
..runkratos: /home/ubuntu/Kratos/external_libraries/rapidjson/document.h:891: rapidjson::GenericValue<Encoding, Allocator>::MemberIterator rapidjson::GenericValue<Encoding, Allocator>::MemberEnd() [with Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>; rapidjson::GenericValue<Encoding, Allocator>::MemberIterator = rapidjson::GenericMemberIterator<false, rapidjson::UTF8<>, rapidjson::MemoryPoolAllocator<> >]: Assertion 'IsObject()' failed.`

I havent commited to Master in months, and it was working. On my machine it still does.
It fails since 10.October, and the only merge to master on the previous day was #722 as far as I could see (which shouldn't have to do anything with this)

Help is appreciated :D

**Comment by philbucher** [comment]
@roigcarlo since `FullDebug` is available you use it for the compilation on the testmachine, right?
Then it makes sense, #722 introduced this, and with FullDebug I also get the error.
I haven't found it yet, but I think I can figure it out, so closing

**Comment by roigcarlo** [comment]
@philbucher  No, test machine is not using `FullDebug`, but is not defining -DNDEBUG, which can lead to similar effects. We have been doing this quite before #722. There were some days just after that PR that it was compiling using -DNDEBUG, so if you committed during those the issue may have been hidden.

For the record, the only reason for this is that `Release` takes to long to compile, but maybe is a good idea to leave like this if it helps to detect things

**Comment by philbucher** [comment]
That is weird.
From the [history](https://github.com/KratosMultiphysics/Kratos/commits/master/applications/MappingApplication) there are broken commits on 13.10. and 14.10. Could those have changed anything? (I think you fixed this, didn't you?)
Other than that, the other commits are mostly cleanup and not related to any rapidjson stuff as far as I could see.
Unfortunately also gdb doesn't point me to a location in the code, so I guess I have to spend more time on this :/

**Comment by roigcarlo** [comment]
Urg those commits... yep let me see I touched something I shouldn't have. Fixing that was quite tough

**Comment by jcotela** [comment]
Let's keep this open until we know what's going on ;)

**Comment by philbucher** [comment]
thx @roigcarlo

**Comment by roigcarlo** [comment]
Hi, @philbucher 

Still not sure why this started to fail on Oct - 10 but I have found the cause and a possible fix. 
It seems that the problem is in this function:

```Python
def AssignPrescribedValues(self, side_interface, direction, value_type, dictionary):

# Some Intersting code
 
for x in coordinates_x:
    # Retreive Coordinates
    coords = (coordinates_x[i].GetDouble(), 
                    coordinates_y[i].GetDouble(), 
                    coordinates_z[i].GetDouble())
```

The problem is that `coordinates_x` seems to have problem being iterable. As I said, no clue why. but as you don't use the iterability property feature, it can be rewritten to:

```Python
for _ in range(0,coordinates_x.size()):
            # Retreive Coordinates
            coords = (coordinates_x[i].GetDouble(), 
                            coordinates_y[i].GetDouble(), 
                            coordinates_z[i].GetDouble())
```

Which prevent the error from happening.

Anyway, i'll keep digging into why this only happens in debug. Looks like the spike of a deeper problem...

**Comment by philbucher** [comment]
Thx a lot @roigcarlo for your help, I will change that

**Comment by philbucher** [comment]
fixed in #1153
